src/psyclone/tests/utilities_test.py

```diff
@@ -49,99 +56,237 @@
 '''
 
 
-def test_compiler_works(tmpdir):
-    ''' Check that the specified compiler works for a hello-world
-    example '''
+def test_enable_disable_compilation(monkeypatch):
+    '''Test the behaviour when disabling compilation ... even if
+    compilation is enabled.
+    '''
+    monkeypatch.setattr(Compile, "TEST_COMPILE", False)
+    # Test that compile_file will do nothing if compilation is disabled:
+    _compile = Compile()
+    _compile.compile_file("nothing-to-compile")
+    _compile.code_compiles(None)
+    _compile.string_compiles("")
     Compile.skip_if_compilation_disabled()
 
-    old_pwd = tmpdir.chdir()
-    try:
-        with open("hello_world.f90", "w", encoding="utf-8") as ffile:
-            ffile.write(HELLO_CODE)
-        Compile(tmpdir).compile_file("hello_world.f90", link=True)
-    finally:
-        old_pwd.chdir()
+
+# -----------------------------------------------------------------------------
+def test_enable_disable_opencl_compilation(monkeypatch):
+    '''Test the behaviour when disabling opencl compilation ... even if
+    compilation is enabled.
+    '''
+    monkeypatch.setattr(Compile, "TEST_COMPILE_OPENCL", False)
+    Compile.skip_if_opencl_compilation_disabled()
```

Comment:
Again, I'm not sure what this actually tests. Please could you add a comment?

Reply:
As above, I've improved this test.